vector/src/main/java/im/vector/app/features/settings/VectorSettingsGeneralFragment.kt

```diff
@@ -450,7 +494,67 @@ class VectorSettingsGeneralFragment @Inject constructor(
                     }
                 }
             }
-            dialog.show()
+
+            TchapSettingsChangePasswordPreDialog(session)
+                    .apply {
+                        listener = object : TchapSettingsChangePasswordPreDialog.InteractionListener {
+                            override fun changePassword() {
+                                dialog.show()
+                            }
+
+                            override fun exportKeys(passphrase: String, uri: Uri) {
+                                showLoadingView(true)
+                                lifecycleScope.launch {
+                                    try {
+                                        keysExporter.export(passphrase, uri)
+                                        requireActivity().toast(getString(R.string.encryption_exported_successfully))
+                                    } catch (failure: Throwable) {
+                                        requireActivity().toast(errorFormatter.toHumanReadable(failure))
+                                    }
+                                    showLoadingView(false)
+                                }
+                            }
+                        }
+                    }
+                    .show(activity.supportFragmentManager, "changePasswordPreDialog")
+        }
+    }
+
+    private fun onHideFromUsersDirectoryClick() {
+        val hidden = hideFromUsersDirectoryPreference.isChecked
+        // The external users must be prompted before showing them to the users directory
+        if (!hidden && TchapUtils.isExternalTchapUser(session.myUserId)) {
+            MaterialAlertDialogBuilder(requireActivity())
+                    .setMessage(R.string.tchap_settings_show_external_user_in_users_directory_prompt)
+                    .setPositiveButton(R.string.accept) { _, _ ->
+                        hideUserFromUsersDirectory(false)
+                    }
+                    .setNegativeButton(R.string.cancel) { _, _ ->
+                        hideFromUsersDirectoryPreference.isChecked = true
+                    }
+                    .setOnCancelListener { _ ->
+                        hideFromUsersDirectoryPreference.isChecked = true
+                    }
+                    .show()
+        } else {
+            hideUserFromUsersDirectory(hidden)
+        }
+    }
+
+    private fun hideUserFromUsersDirectory(hidden: Boolean) {
+        displayLoadingView()
+        lifecycleScope.launch {
+            val result = runCatching { session.accountDataService().updateHideProfile(hidden) }
```

Comment:
You should not create a new API in the SDK, but rather use the existing API `session.accountDataService().updateUserAccountData(...)`

Reply:
we agree with this remark, @yostyle will update the PR